reframe/core/schedulers/__init__.py

```diff
@@ -222,12 +222,22 @@ def sched_exclusive_access(self):
     def prepare(self, commands, environs=None, **gen_opts):
         environs = environs or []
         if self.num_tasks <= 0:
+            num_tasks_per_node = (self.num_tasks_per_node if
+                                  self.num_tasks_per_node else 1)
+            min_num_tasks = (abs(self.num_tasks) if self.num_tasks < 0 else
```

Comment:
You know it is negative, so you can simply use `-self.num_tasks`. You might be able to fit this in a single line if you rewrite it as follows:

```python
min_num_tasks = -self.num_tasks if self.num_tasks else num_tasks_per_node
```